vispy/color/colormap.py

```diff
@@ -116,44 +119,78 @@ def step(colors, x, controls=None):
 
 
 # GLSL interpolation functions.
-def _glsl_mix(controls=None):
+def _glsl_mix(controls=None, colors=None, texture_map_data=None):
     """Generate a GLSL template function from a given interpolation patterns
-    and control points."""
+    and control points.
+
+    Parameters
+    ----------
+    colors : array-like, shape (n_colors, 4)
+        The control colors used by the colormap.
+        Elements of colors must be convertible to an instance of Color-class.
+
+    controls : list
+        The list of control points for the given colors. It should be
+        an increasing list of floating-point number between 0.0 and 1.0.
+        The first control point must be 0.0. The last control point must be
+        1.0. The number of control points depends on the interpolation scheme.
+
+    texture_map_data : ndarray, shape(LUT_len, 4)
+        Numpy array of size of 1D texture lookup data
+        for luminance to RGBA conversion.
+
+    """
     assert (controls[0], controls[-1]) == (0., 1.)
     ncolors = len(controls)
     assert ncolors >= 2
-    if ncolors == 2:
-        s = "    return mix($color_0, $color_1, t);\n"
-    else:
-        s = ""
-        for i in range(ncolors-1):
-            if i == 0:
-                ifs = 'if (t < %.6f)' % (controls[i+1])
-            elif i == (ncolors-2):
-                ifs = 'else'
-            else:
-                ifs = 'else if (t < %.6f)' % (controls[i+1])
-            adj_t = '(t - %s) / %s' % (controls[i],
-                                       controls[i+1] - controls[i])
-            s += ("%s {\n    return mix($color_%d, $color_%d, %s);\n} " %
-                  (ifs, i, i+1, adj_t))
-    return "vec4 colormap(float t) {\n%s\n}" % s
-
-
-def _glsl_step(controls=None):
+    assert (texture_map_data is not None)
+
+    LUT = texture_map_data
+    LUT_len = texture_map_data.shape[0]
```

Comment:
this shadows the global variable, one of them should have a different name for code clarity